src/vs/workbench/services/userDataProfile/browser/extensionsResource.ts

```diff
@@ -48,8 +48,10 @@ export class ExtensionsResource implements IProfileResource {
 
 	async apply(content: string, profile: IUserDataProfile): Promise<void> {
 		return this.withProfileScopedServices(profile, async (extensionEnablementService) => {
-			const profileExtensions: IProfileExtension[] = await this.getProfileExtensions(content);
-			const installedExtensions = await this.extensionManagementService.getInstalled(undefined, profile.extensionsResource);
+			const [profileExtensions, installedExtensions] = await Promise.all([
+				this.getProfileExtensions(content),
+				this.extensionManagementService.getInstalled(undefined, profile.extensionsResource)
+			]) as [IProfileExtension[]];
```

Comment:
Why do you have this casting?

Reply:
Sorry, I thought this would add type support :(

If it is useless I can remove it 🙌

Reply:
> Why do you have this casting?

Sorry, I thought this would add type support :(

If it is useless I can remove it 🙌